desktop/notification/manager_test.go

```diff
@@ -0,0 +1,78 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package notification_test
+
+import (
+	"fmt"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/godbus/dbus"
+	"github.com/snapcore/snapd/desktop/notification"
+	"github.com/snapcore/snapd/testutil"
+)
+
+type managerSuite struct {
+	testutil.BaseTest
+	testutil.DBusTest
+}
+
+var _ = Suite(&managerSuite{})
+
+func (s *managerSuite) SetUpTest(c *C) {
+}
+
+func (s *managerSuite) TestUseGtkBackendIfAvailable(c *C) {
+	restoreGtk := notification.MockNewGtkBackend(func(conn *dbus.Conn, desktopID string) (notification.NotificationManager, error) {
+		return &notification.GtkBackend{}, nil
+	})
+	defer restoreGtk()
+
+	restoreFdo := notification.MockNewFdoBackend(func(conn *dbus.Conn, desktopID string) notification.NotificationManager {
+		c.Fatalf("fdo backend shouldn't be created")
+		return nil
+	})
+	defer restoreFdo()
+
+	mgr := notification.NewNotificationManager(s.SessionBus, "desktop-id")
+	c.Assert(mgr, NotNil)
```

Comment:
I would change this into a `Check`, and add a line below where you check that `mgr` is actually the same `&notification.GtkBackend{}` you injected in the gtk mock (then you'll have to create it at the beginning of the test, before the call to `MockNewGtkBackend`.